packages/core/src/axis-chart.ts

```diff
@@ -60,23 +60,27 @@ export class AxisChart extends Chart {
 			}
 		};
 
+		const isEnabled = this.model.getOptions().legend.enabled !== false;
+
 		// Decide the position of the legend in reference to the chart
 		let fullFrameComponentDirection = LayoutDirection.COLUMN;
-		const legendPosition = Tools.getProperty(this.model.getOptions(), "legend", "position");
-		if (legendPosition === LegendPositions.LEFT) {
-			fullFrameComponentDirection = LayoutDirection.ROW;
+		if (isEnabled) {
```

Comment:
Also `isEnabled` is a misleading var name considering that this is not the legend component. We should've used something more descriptive such as `isLegendEnabled`

Reply:
@theiliad is it possible to integrate this as is for the near term problem then migrate that code to Legend along with the Chart class changes?

Reply:
I wouldn't want to do that. Moving stuff into the legend component shouldn't take long. The other chart class changes don't yet affect us, and I will take care of those changes